ddtrace/tracer/context_test.go

```diff
@@ -107,6 +110,34 @@ func TestStartSpanFromContextRace(t *testing.T) {
 	assert.ElementsMatch(t, outputs, expectedTraceIDs)
 }
 
+func Test128(t *testing.T) {
+	_, _, _, stop := startTestTracer(t)
+	defer stop()
+
+	span, _ := StartSpanFromContext(context.Background(), "http.request")
+	assert.NotZero(t, span.Context().TraceID())
+	if w3cCtx, ok := span.Context().(ddtrace.SpanContextW3C); ok {
+		assert.Empty(t, w3cCtx.TraceID128())
+	} else {
+		assert.Fail(t, "should be castable to SpanContextW3C")
+	}
+
+	// Enable 128 bit trace ids
+	t.Setenv("DD_TRACE_128_BIT_TRACEID_ENABLED", "true")
+	span128, _ := StartSpanFromContext(context.Background(), "http.request")
+	assert.NotZero(t, span128.Context().TraceID())
+	if w3cCtx, ok := span128.Context().(ddtrace.SpanContextW3C); ok {
+		// Ensure that the lower order bits match the span's 64-bit trace id
+		id128bit := w3cCtx.TraceID128()
+		assert.NotEmpty(t, id128bit)
+		b, err := hex.DecodeString(id128bit)
+		assert.NoError(t, err)
+		assert.Equal(t, span128.Context().TraceID(), binary.BigEndian.Uint64(b[8:]))
```

Comment:
Are the expectation and actual flipped here?

Reply:
Maybe? I'm making sure that the lower bits in the `SpanContextW3C.TraceID128` (actual) have the expected value of `SpanContext.TraceID`. It's just making sure they're the same, but which is actual and which is expected is kind of up for debate 🤷